lib/inspec/plugins/resource.rb

```diff
@@ -65,6 +71,14 @@ def skip_resource(message)
           @resource_skipped = message
         end
 
+        def resource_failed
+          @resource_failed if defined?(@resource_failed)
```

Comment:
If `@resource_failed` is never assigned a value, it will return `nil` by default. Do we need this `defined?` check?